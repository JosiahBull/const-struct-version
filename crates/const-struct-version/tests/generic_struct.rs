#![allow(dead_code)]

use const_struct_version::StructVersion;

#[derive(StructVersion)]
struct GenericContainer<T> {
    value: T,
    items: Vec<T>,
}

#[derive(StructVersion)]
struct ConcreteType {
    data: GenericContainer<u32>,
}

#[test]
fn test_generic_struct() {
    let version = <ConcreteType as StructVersion>::version();
    insta::assert_debug_snapshot!(version);
}
