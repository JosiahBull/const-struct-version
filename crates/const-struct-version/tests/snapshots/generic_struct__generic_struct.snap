---
source: crates/const-struct-version/tests/generic_struct.rs
expression: version
---
"29e58e20ca9432024b9fd41a3998863d5aa47c41"
